source/Halibut/HalibutRuntime.cs

```diff
@@ -74,6 +78,7 @@ public HalibutRuntime(IServiceFactory serviceFactory, X509Certificate2 serverCer
             invoker = new ServiceInvoker(serviceFactory);
 
             connectionManager = new ConnectionManager();
+            this.tcpConnectionFactory = new TcpConnectionFactory(serverCertificate, AsyncHalibutFeature, TimeoutsAndLimits, streamFactory);
```

Comment:
I would have avoided this change. Yes it makes it cleaner in some areas but it's starting to diverge from our attempt to make as few changes as possible to sync halibut

Reply:
Yeah I was worried about that as well, however its such a straight forward top level change that I become less concerned. The change felt worth it since it just simplified the code by a decent amount.